Build is currently broken

```
...:coda-prototype lukechurch$ tsc
src/io/FileUtils.ts(1,1): error TS6054: File '/Users/lukechurch/GitRepos/coda/coda-prototype/UIUtils.js' has unsupported extension. The only supported extensions are '.ts', '.tsx', '.d.ts'.
src/io/FileUtils.ts(88,34): error TS2304: Cannot find name 'Papa'.
src/io/FileUtils.ts(119,34): error TS2304: Cannot find name 'Papa'.
src/io/FileUtils.ts(153,29): error TS2304: Cannot find name 'Papa'.
src/io/FileUtils.ts(275,29): error TS2304: Cannot find name 'Papa'.
src/io/FileUtils.ts(323,43): error TS2304: Cannot find name 'UIUtils'.
```

--- a/coda-prototype/.gitignore
+++ b/coda-prototype/.gitignore
@@ -1,0 +1,3 @@
+.idea/
+node_modules/
+coda.zip

--- a/coda-prototype/deploy.sh
+++ b/coda-prototype/deploy.sh
@@ -1,31 +1,29 @@
 #!/bin/bash
-set -e;
+set -e
 
-# Delete last attempt if it exists
-rm -rf coda;
+# Revert version.json on exit.
+function finish {
+    echo "{\"hash\": \"develop\", \"date\": \"\"}" >version.json
+}
+trap finish EXIT
 
-# Create a temporary directory to export to
-mkdir coda;
+# If the current directory does not match the HEAD commit, emit a warning,
+# otherwise export the hash and date of HEAD to version.json
+changes=$(git status --porcelain) # Machine readable list of changes; empty if there no changes.
+if [ ! -z "${changes}" ]; then
+    echo "Warning: File system is different to HEAD, so the zipped project will not
+         include a version identifier."
+else
+    # Get the commit hash and time-stamp
+    hash=$(git rev-parse HEAD)
+    date=$(git show -s --format=%ci HEAD)
 
-# Check out HEAD to the temporary directory.
-# Export via git archive, which exports as a zip which we have to unzip.
-git archive HEAD --output=coda/export.zip;
-unzip -qq coda/export.zip -d coda/;
-rm coda/export.zip;
+    # Write the hash and time-stamp to a file
+    echo "{\"hash\": \"$hash\", \"date\": \"$date\"}" >version.json
+fi
 
-# Build project
-# Note: will need to run npm install here in future.
-tsc;
+# Delete the previously deployed version, if it exists, so that it is not included in the next zip file.
+[ -e coda.zip ] && rm coda.zip
 
-# Get the commit hash and time-stamp
-hash=$(git rev-parse HEAD);
-date=$(git show -s --format=%ci HEAD)
-
-# Write the hash and time-stamp to a file
-echo "{\"hash\": \"$hash\", \"date\": \"$date\"}" >coda/version.json;
-
-# Zip up the output
-zip -r -qq coda.zip coda/;
-
-# Clean-up the temporary export directory.
-rm -rf coda;
+# Zip this directory into coda.zip
+zip -r -qq coda.zip .

--- a/coda-prototype/deploy.sh
+++ b/coda-prototype/deploy.sh
@@ -3,7 +3,7 @@ set -e
 
 # Revert version.json on exit.
 function finish {
-    echo "{\"hash\": \"develop\", \"date\": \"\"}" >version.json
+    echo $'{\n  "hash": "develop",\n  "date": ""\n}' >version.json
 }
 trap finish EXIT
 

--- a/coda-prototype/version.json
+++ b/coda-prototype/version.json
@@ -2,4 +2,3 @@
   "hash": "develop",
   "date": ""
 }
-

